### PR TITLE
Added even more lua functions

### PR DESCRIPTION
**Pull request by abcdefg30**
This inculdes functions to:
- set beacons
- enter transports
- ~~demolish actors~~ #8576

### CODE REVIEW

File: OpenRA.Mods.Common/Scripting/Global/BeaconGlobal.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	public class BeaconGlobal : ScriptGlobal
+	{
+		public BeaconGlobal(ScriptContext context) : base(context) { }
+
+		[Desc("Creates a new beacon that stays for the specified time at the specified WPos." +
+			"Does not remove player set beacons, nor gets removed by placing them.")]
+		public Beacon New(Player owner,  WPos position, int duration = 30 * 25, string palettePrefix = "player")
+		{
+			var playerBeacon = new Beacon(owner, position, duration, palettePrefix);
+			owner.PlayerActor.World.AddFrameEndTask(w => w.Add(playerBeacon));
```
Comment (pchote) [inline OpenRA.Mods.Common/Scripting/Global/BeaconGlobal.cs]: This will also miss the radar ping and "press spacebar to go to beacon" input.

Comment (abcdefg30) [inline OpenRA.Mods.Common/Scripting/Global/BeaconGlobal.cs]: Hm, I should fix the radar ping, but do we want "press spacebar to go to beacon" (as there could be multiple beacons and just jumping to the last one sounds unpolished to me)?

**Comment by Mailaender** [comment]
Possible test cases:
* Demolition: the jeep in the beginning of soviet-01
* Beacon: allies-02 when the trucks arrive.

File: OpenRA.Mods.Common/Scripting/Global/BeaconGlobal.cs (hunk 1 above)
Comment (abcdefg30) [inline OpenRA.Mods.Common/Scripting/Global/BeaconGlobal.cs]: Whatever, you are able to use the spacebar jump when using the radar pings.

**Comment by abcdefg30** [comment]
Updated.

> Possible test cases:
- Demolition: the jeep in the beginning of soviet-01
- Beacon: allies-02 when the trucks arrive.

- The bridge is actually blown up by the barrel explosion there. I think using the jeep to blow it up would be a bigger hack then just calling `bridge.Kill()`.
- Done (also in survival01).

File: mods/ra/maps/allies-02/allies02.lua [hunk 2]
```diff
@@ -70,6 +70,7 @@ SendTrucks = function()
 	if not ConvoyOnSite then
 		ConvoyOnSite = true
 		ticked = 0
+		Beacon.New(player, TruckEntryPoint.CenterPosition + WVec.New(0, 3 * 1024, 0))
```
Comment (obrakmann) [inline mods/ra/maps/allies-02/allies02.lua]: This is ok as a testcase, but I think a bit pointless without a radar.

**Comment by obrakmann** [comment]
besides that, :+1:

**Comment by abcdefg30** [comment]
Updated. Removed the beacon from allies02 and added one to survival02.

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj [hunk 3]
```diff
@@ -230,6 +231,7 @@
     <Compile Include="Scripting\Properties\GuardProperties.cs" />
     <Compile Include="Scripting\Properties\MissionObjectiveProperties.cs" />
     <Compile Include="Scripting\Properties\MobileProperties.cs" />
+    <Compile Include="Scripting\Properties\DemolitionProperties.cs" />
```
Comment (Mailaender) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: Then again just put this into a separate pull request so we can merge it later with a test case.

Comment (abcdefg30) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: Done: #8576

**Comment by Mailaender** [comment]
Works as promised. :white_check_mark: Again, nice expansions to the Lua API.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/da62830b04488d0e2e62abc2d1a0f48814128ffb...78002d14d8191d3f42e701733a5bb0d0a9c51467)

**Pull request merged**